product_pack/models/product_template.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2019 Tecnativa - Ernesto Tejeda
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import fields, models, api, _
+from odoo.exceptions import ValidationError
+
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    pack_type = fields.Selection([
+        ('components_price', 'Detailed - Components Prices'),
+        ('totaliced_price', 'Detailed - Totaliced Price'),
+        ('fixed_price', 'Detailed - Fixed Price'),
+        ('none_detailed_assisted_price', 'None Detailed - Assisted Price'),
+        ('none_detailed_totaliced_price', 'None Detailed - Totaliced Price'),
+    ],
+        'Pack Type',
+        help="On sale orders or purchase orders:\n"
+             "* Detailed - Components Prices: Detail lines with prices.\n"
+             "* Detailed - Totaliced Price: Detail lines totalicing "
+             "lines prices on pack (don't show component prices).\n"
+             "* Detailed - Fixed Price: Detail lines and use product"
+             " pack price (ignore line prices).\n"
+             "* None Detailed - Assisted Price: Do not detail lines. "
+             "Assist to get pack price using pack lines./n"
+             "* None Detailed - Totaliced Price: Do not detail lines. "
+             "Do not assist to get pack price using pack lines."
+    )
+    pack = fields.Boolean(
+        'Pack?',
+        help='Is a Product Pack?',
+    )
+    pack_line_ids = fields.One2many(
+        related='product_variant_ids.pack_line_ids',
+        readonly=True,
+    )
+    used_pack_line_ids = fields.One2many(
+        related='product_variant_ids.used_pack_line_ids',
+        readonly=True,
+    )
+    allow_modify_pack = fields.Boolean()
+
+    @api.onchange('pack_type')
+    def onchange_allow_modify_pack(self):
+        products = self.filtered(
+            lambda x: x.allow_modify_pack and
+            x.pack_type and x.pack_type != 'components_price')
+        for rec in products:
+            rec.allow_modify_pack = False
+
+    @api.constrains('pack_type')
+    def check_relations(self):
+        """
+        Check assisted packs don't have packs a child
```

Comment:
```suggestion
        """Check assisted packs don't have packs a child."""
```

See https://www.python.org/dev/peps/pep-0257/#one-line-docstrings